plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/CubridTable.java

```diff
@@ -33,86 +33,224 @@
 import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
 import org.jkiss.dbeaver.model.struct.DBSObject;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.List;
 
 public class CubridTable extends GenericTable
 {
+    private CubridUser owner;
+    private CubridUser oldOwner;
+    private CubridCharset charset;
+    private CubridCollation collation;
+    private Integer autoIncrement;
+    private boolean reuseOID = true;
+
     public CubridTable(
-            GenericStructContainer container,
+            @NotNull GenericStructContainer container,
             @Nullable String tableName,
             @Nullable String tableType,
-            @Nullable JDBCResultSet dbResult)
-    {
+            @Nullable JDBCResultSet dbResult) {
         super(container, tableName, tableType, dbResult);
 
+        String ownerName;
+        String collationName;
         if (dbResult != null) {
             String type = JDBCUtils.safeGetString(dbResult, CubridConstants.IS_SYSTEM_CLASS);
+            this.reuseOID = (JDBCUtils.safeGetString(dbResult, CubridConstants.REUSE_OID)).equals("YES");
+            ownerName = JDBCUtils.safeGetString(dbResult, CubridConstants.OWNER_NAME);
+            collationName = JDBCUtils.safeGetString(dbResult, CubridConstants.COLLATION);
+            autoIncrement = JDBCUtils.safeGetInteger(dbResult, CubridConstants.AUTO_INCREMENT_VAL);
             if (type != null) {
                 this.setSystem(type.equals("YES"));
             }
+        } else {
+            ownerName = getDataSource().getContainer().getConnectionConfiguration().getUserName();
+            ownerName = ownerName != null ? ownerName.toUpperCase() : "";
+            collationName = CubridConstants.DEFAULT_COLLATION;
+        }
+
+        for(GenericSchema cubridOwner : this.getDataSource().getSchemaList()){
+            if(cubridOwner.getName().equals(ownerName)) {
+                this.owner = (CubridUser) cubridOwner;
+                this.oldOwner = (CubridUser) cubridOwner;
+            }
+        }
+
+        for(CubridCharset cbCharset : getDataSource().getCharsets()){
+            if(cbCharset.getName().equals(collationName.split("_")[0])) {
+                this.charset = cbCharset;

```

Comment:
`collationName.split("_")[0]` should be in a variable.

Reply:
Thanks for suggestion. I have fixed this.